.circleci/config.yml

```diff
@@ -16,17 +21,97 @@ jobs:
             git submodule update
       - setup_remote_docker
       - run:
-          name: Build Docker image
-          command: docker build -t prio .
+          name: Create a version.json
+          command: |
+            # create a version.json per https://github.com/mozilla-services/Dockerflow/blob/master/docs/version_object.md
+            printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n' \
+            "$CIRCLE_SHA1" \
+            "$CIRCLE_TAG" \
+            "$CIRCLE_PROJECT_USERNAME" \
+            "$CIRCLE_PROJECT_REPONAME" \
+            "$CIRCLE_BUILD_URL" > version.json
+      - run:
+          name: Build staged images
+          command: |
+            docker build --target development -t prio:dev .
+            docker build --target production -t prio:prod .
+      - run:
+          name: Save image into cache
+          command: |
+            mkdir -p /cache
+            docker save -o /cache/dev.tar "prio:dev"
+            docker save -o /cache/prod.tar "prio:prod"
+      - save_cache:
+          key: v1-{{ .Branch }}-{{ epoch }}
+          paths:
+            - /cache/dev.tar
+            - /cache/prod.tar
```

Comment:
The verbosity of this stuff always makes me sad :(

Reply:
On the other hand, it's very explicit about everything. I think it's pretty nice to have control over the CI workflow.